pkg/controller/oneagent/oneagent_utils.go

```diff
@@ -85,6 +86,7 @@ func copyDaemonSetSpecToOneAgentSpec(dsSpec *appsv1.DaemonSetSpec, crSpec *dynat
 	}
 	crSpec.PriorityClassName = dsSpec.Template.Spec.PriorityClassName
 	crSpec.DNSPolicy = dsSpec.Template.Spec.DNSPolicy
+	crSpec.Labels = dsSpec.Template.Labels
```

Comment:
I've the strong impression that this will force the Operator to update the DS on every reconciliation, because the DS labels won't match what it's on CR. Look at `hasSpecChanged()` for implementation details.

I suggest leaving this line here, but moving the labels generation from `newDaemonSetForCR()` to `SetDefaults_OneAgentSpec()` in `default.go`.

Reply:
Since `SetDefaults_OneAgentSpec()` doesn't seem to work since the name is not available, then I suggest `ReconcileOneAgent.Reconcile()` instead, as we are doing with the tokens already.